src/app/kbv/controllers/single-amount-question.test.js

```diff
@@ -0,0 +1,156 @@
+      questionKey: "Q1",
+      hint: "l",
+      text: "t",
+    };
+  });
+
+  it("should be an instance of BaseController", () => {
+    controller = new Controller({ route: "/test" });
+    expect(controller).toBeInstanceOf(BaseController);
+  });
+
+  describe("#locals", () => {
+    beforeEach(() => {
+      req.session = {
+        question: {},
+      };
+      req.translate = jest.fn();
+    });
+
+    it("should set threeMonthsAgo in locals", (done) => {
+      const callback = jest.fn((err, locals) => {
+        expect(err).toBeNull();
+        expect(locals.threeMonthsAgo).toEqual(
+          moment().subtract(3, "months").format("DD MMMM YYYY")
+        );
+        done();
+      });
+
+      controller.locals(req, res, callback);
+    });
+
+    it("should call super.locals with req and res", () => {
+      const superLocals = jest.spyOn(BaseController.prototype, "locals");
+
+      controller.locals(req, res, next);
+
+      expect(superLocals).toHaveBeenCalledWith(req, res, expect.any(Function));
+    });
+
+    it("should set question.label and question.hint in locals", (done) => {
+      presenters.questionToLabel.mockReturnValue("Question Label");
+      presenters.questionToHint.mockReturnValue("Question Hint");
+
+      const callback = jest.fn((err, locals) => {
+        expect(err).toBeNull();
+        expect(locals.question.label).toBe("Question Label");
+        expect(locals.question.hint).toBe("Question Hint");
+        done();
+      });
+
+      controller.locals(req, res, callback);
+    });
+
+    it("should call callback with error if super.locals returns an error", (done) => {
+      const mockError = new Error("Some error");
+
+      const callback = jest.fn((err, locals) => {
+        expect(err).toEqual(mockError);
+        expect(locals).toBeUndefined();
+        done();
+      });
+
+      const superLocals = jest.spyOn(BaseController.prototype, "locals");
+      superLocals.mockImplementation((req, res, callback) => {
+        callback(mockError);
+      });
+
+      controller.locals(req, res, callback);
+    });
+  });
+
+  describe("#saveValues", () => {
+    beforeEach(() => {
+      req.session.tokenId = "session-id";
+      req.axios.post = jest.fn();
+    });
+    it("should call single-amount-question endpoint", async () => {
+      await controller.saveValues(req, res, next);
+
+      expect(req.axios.get).toHaveBeenCalledWith(QUESTION, {
+        headers: {
+          "session-id": req.session.tokenId,
+          session_id: req.session.tokenId,
+        },
+      });
+    });
+
+    describe("on API success", () => {
+      it("should call post and get api", async () => {
+        req.axios.get = jest.fn().mockResolvedValue({});
+        req.axios.post = jest.fn().mockResolvedValue({});
+
+        await controller.saveValues(req, res, next);
+
+        expect(req.axios.post).toHaveBeenCalledTimes(1);
+        expect(req.axios.get).toHaveBeenCalledTimes(1);
+      });
+    });
```

Comment:
Could we have just 2 test cases
- should call answer endpoint to post submitted answer
- should call question endpoint to get next question and store it in session
with each checking the specific calls, what we pass to post and get